src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlVirtualDocument.cs

```diff
@@ -23,11 +25,26 @@ public HtmlVirtualDocument(Uri uri, ITextBuffer textBuffer)
             }
 
             Uri = uri;
-            _textBuffer = textBuffer;
+            TextBuffer = textBuffer;
+            UpdateSnapshot();
         }
 
         public override Uri Uri { get; }
 
-        public override long? HostDocumentSyncVersion => throw new NotImplementedException();
+        public override ITextBuffer TextBuffer { get; }
+
+        public override long? HostDocumentSyncVersion => null;
+
+        public override VirtualDocumentSnapshot CurrentSnapshot => _currentSnapshot;
+
+        public override VirtualDocumentSnapshot Update(IReadOnlyList<TextChange> changes, long hostDocumentVersion)
+        {
+            throw new NotImplementedException();
```

Comment:
:sad_trombone:?

Reply:
@ajaybhargavb will be adding the HTML update support.